src/main/java/org/breedinginsight/brapps/importer/services/processors/experiment/create/workflow/CreateNewExperimentWorkflow.java

```diff
@@ -0,0 +1,304 @@
+        this.populateNewPendingImportObjectsStep = populateNewPendingImportObjectsStep;
+        this.commitPendingImportObjectsStep = commitPendingImportObjectsStep;
+        this.validatePendingImportObjectsStep = validatePendingImportObjectsStep;
+        this.statusService = statusService;
+        this.experimentPhenotypeService = experimentPhenotypeService;
+        this.workflow = ExperimentWorkflowNavigator.Workflow.NEW_OBSERVATION;
+    }
+
+    private ImportPreviewResponse runWorkflow(ImportContext context) throws Exception {
+
+        ImportUpload upload = context.getUpload();
+        boolean commit = context.isCommit();
+        List<BrAPIImport> importRows = context.getImportRows();
+
+        // Make sure the file does not contain obs unit ids before proceeding
+        if (containsObsUnitIDs(context)) {
+            throw new HttpStatusException(HttpStatus.UNPROCESSABLE_ENTITY, "ObsUnitIDs are detected");
+        }
+
+        statusService.updateMessage(upload, "Checking existing experiment objects in brapi service and mapping data");
+
+        ProcessedPhenotypeData phenotypeData = experimentPhenotypeService.extractPhenotypes(context);
+        ProcessContext processContext = populateExistingPendingImportObjectsStep.process(context, phenotypeData);
+        ProcessedData processedData = populateNewPendingImportObjectsStep.process(processContext, phenotypeData);
+        ValidationErrors validationErrors = validatePendingImportObjectsStep.process(context, processContext.getPendingData(), phenotypeData, processedData);
+
+        // short circuit if there were validation errors
+        if (validationErrors.hasErrors()) {
+            throw new ValidatorException(validationErrors);
+        }
+
+        // TODO: move to experiment import service
+        ImportPreviewResponse response = buildImportPreviewResponse(importRows, processContext.getPendingData(), processedData, upload);
+
+        statusService.updateMappedData(upload, response, "Finished mapping data to brapi objects");
+
+        // preview data
+        if (!commit) {
+            statusService.updateOk(upload);
+            return response;
+        }
+
+        // commit data
+        long totalObjects = getNewObjectCount(response);
+        statusService.startUpload(upload, totalObjects, "Starting upload to brapi service");
+        statusService.updateMessage(upload, "Creating new experiment objects in brapi service");
+
+        commitPendingImportObjectsStep.process(processContext, processedData);
+
+        statusService.finishUpload(upload, totalObjects, "Completed upload to brapi service");
+        return response;
+    }
+
+    /**
+     * Retrieves the name of the workflow. This is used for logging display purposes.
+     *
+     * @return the name of the workflow
+     */
+    public Optional<ImportWorkflowResult> process(ImportServiceContext context) throws Exception {
```

Comment:
I think this javadoc comment belongs to a different method.

Reply:
Thanks ya looks like something from merging with the updated workflow selection code. Updated comment and pushed changes.